generic-server/src/plugin/container.rs

```diff
@@ -185,87 +192,143 @@ mod tests {
 
     #[test]
     fn test_loading() {
+        // Mock plugins for testing
+        let plugins: Arc<Vec<Arc<Mutex<dyn Plugin>>>> = Arc::new(vec![
+            Arc::new(Mutex::new(FirstPlugin {})),
+            Arc::new(Mutex::new(SecondPlugin {})),
+        ]);
+
+        // Initialize PluginContainer with the mock plugins
         let mut container = PluginContainer {
             loaded: false,
             collected_routes: vec![],
-            plugins: &vec![Box::new(FirstPlugin {}), Box::new(SecondPlugin {})],
+            plugins: Arc::clone(&plugins),
         };
 
+        // Test loading plugins
         assert!(container.load().is_ok());
         assert!(container.routes().is_ok());
 
+        // Verify find_plugin method
         assert!(container.find_plugin("first").is_some());
         assert!(container.find_plugin("second").is_some());
         assert!(container.find_plugin("non-existent").is_none());
 
-        // The actual routes collected are actually hard to test
-        // given that axum::Router seems not to provide public
-        // directives to inquire internal state.
-        // See: https://github.com/tokio-rs/axum/discussions/860
+        // Verify collected routes
```

Comment:
Do not remove the extensive comment that was there.
```suggestion
        // Verify collected routes
        // The actual routes collected are actually hard to test
        // given that axum::Router seems not to provide public
        // directives to inquire internal state.
        // See: https://github.com/tokio-rs/axum/discussions/860
```